src/lib/alloc.c

```diff
@@ -167,6 +168,10 @@ static void *rmalloc_sys(struct mm_heap *heap, uint32_t flags, int caps, size_t
 	heap->info.used += bytes;
 	heap->info.free -= alignment + bytes;
 
+	if (is_trace_on())
```

Comment:
This should be managed in the trace layer and not by clients.

Reply:
good point, let me figure it out.